File: user-service/src/logger/summary.ts
import dayjs from 'dayjs'
import logger from './logger'

export interface Summary {
  log_time?: string
  hostname?: string
  appname?: string
  instance?: string
  log_name?: string
  intime: string
  out_time?: string
  diff_time?: number
  ssid?: string
  invoke?: string
  audit_log_id?: string
  mobile_no?: string
  input?: string
  output?: string
  status?: number
  result_code?: string
  command?: string
  menu_id?: string
  channel?: string
}

export function toSummaryLog(newSummaryLog: Summary): void {
  // Parse the intime
  const startDate = dayjs(newSummaryLog.intime)
  const endTime = dayjs()

  // Set fields as per logic
  newSummaryLog.log_name = 'SUMMARY'
  newSummaryLog.log_time = endTime.toISOString()
  newSummaryLog.out_time = endTime.toISOString()
  newSummaryLog.diff_time = endTime.diff(startDate, 'milliseconds')

  // Truncate 'input' and 'output' fields if they exceed 2000 characters
  if (newSummaryLog.input && newSummaryLog.input.length > 2000) {
    newSummaryLog.input = newSummaryLog.input.substring(0, 2000)
  }
  if (newSummaryLog.output && newSummaryLog.output.length > 2000) {
    newSummaryLog.output = newSummaryLog.output.substring(0, 2000)
  }

  newSummaryLog.output = newSummaryLog.output?.replace(/\\"/g, '').replace(/\\\\/g, '\\')

  // Log the summary using the logger library
  logger.info(newSummaryLog)
}

File: go-service/logger/summary.go
package logger

import (
	"encoding/json"
	"time"
)

// type SummaryLogKey struct{}

type Summary struct {
	LogTime    time.Time `json:"log_time"`
	Hostname   string    `json:"hostname"`
	Appname    string    `json:"appname"`
	Instance   string    `json:"instance"`
	LogName    string    `json:"log_name"`
	Intime     string    `json:"in_time"`
	Outtime    string    `json:"out_time"`
	DiffTime   int64     `json:"diff_time"`
	Ssid       string    `json:"ssid"`
	Invoke     string    `json:"invoke"`
	AuditLogId string    `json:"audit_log_id"`
	MobileNo   string    `json:"mobile_no"`
	Input      string    `json:"input"`
	Output     string    `json:"output"`
	Status     int       `json:"status"`
	ResultCode string    `json:"result_code"`
	Command    string    `json:"command"`
	MenuId     string    `json:"menu_id"`
	Channel    string    `json:"channel"`
}

func ToSummaryLog(newSummaryLog Summary) {
	startDate, _ := time.Parse(time.RFC3339, newSummaryLog.Intime)
	endTime := time.Now()
	newSummaryLog.LogName = "SUMMARY"
	newSummaryLog.LogTime = time.Now()
	newSummaryLog.Outtime = endTime.Format(time.RFC3339)
	newSummaryLog.DiffTime = endTime.Sub(startDate).Milliseconds()

	if len(newSummaryLog.Input) > 2000 {
		newSummaryLog.Input = newSummaryLog.Input[0:2000]
	}
	if len(newSummaryLog.Output) > 2000 {
		newSummaryLog.Output = newSummaryLog.Output[0:2000]
	}
	jsonBytes, _ := json.Marshal(newSummaryLog)

	summaryString := string(jsonBytes)

	Logger.Info(summaryString)
}
